src/test/ui/numbers-arithmetic/saturating-float-casts.rs

```diff
@@ -85,11 +89,445 @@ macro_rules! fptoui_tests {
+        /*-0x1p-149*/ f64::from_bits(0xb6a0000000000000),
+    );
+    assert_eq::<f64>(
+        /*0x1.fffffep+127*/ f32::from_bits(0x7f7fffff) as f64,
+        /*0x1.fffffep+127*/ f64::from_bits(0x47efffffe0000000),
+    );
+    assert_eq::<f64>(
+        /*-0x1.fffffep+127*/ (-f32::from_bits(0x7f7fffff)) as f64,
+        /*-0x1.fffffep+127*/ -f64::from_bits(0x47efffffe0000000),
+    );
+    assert_eq::<f64>(
+        /*0x1p-119*/ f32::from_bits(0x4000000) as f64,
+        /*0x1p-119*/ f64::from_bits(0x3880000000000000),
+    );
+    assert_eq::<f64>(
+        /*0x1.8f867ep+125*/ f32::from_bits(0x7e47c33f) as f64,
+        6.6382536710104395e+37,
+    );
+    assert_eq::<f64>(f32::INFINITY as f64, f64::INFINITY);
+    assert_eq::<f64>(f32::NEG_INFINITY as f64, f64::NEG_INFINITY);
+
+    // f64 -> f32
+    assert_eq::<u32>((0.0f64 as f32).to_bits(), 0.0f32.to_bits());
+    assert_eq::<u32>(((-0.0f64) as f32).to_bits(), (-0.0f32).to_bits());
+    assert_eq::<f32>(5.0f64 as f32, 5.0f32);
+    assert_eq::<f32>(/*0x0.0000000000001p-1022*/ f64::from_bits(0x1) as f32, 0.0);
+    assert_eq::<f32>(/*-0x0.0000000000001p-1022*/ (-f64::from_bits(0x1)) as f32, -0.0);
+    assert_eq::<f32>(
+        /*0x1.fffffe0000000p-127*/ f64::from_bits(0x380fffffe0000000) as f32,
+        /*0x1p-149*/ f32::from_bits(0x800000),
+    );
+    assert_eq::<f32>(
+        /*0x1.4eae4f7024c7p+108*/ f64::from_bits(0x46b4eae4f7024c70) as f32,
+        /*0x1.4eae5p+108*/ f32::from_bits(0x75a75728),
+    );
+    assert_eq::<f32>(f64::MAX as f32, f32::INFINITY);
+    assert_eq::<f32>(f64::MIN as f32, f32::NEG_INFINITY);
+    assert_eq::<f32>(f64::INFINITY as f32, f32::INFINITY);
+    assert_eq::<f32>(f64::NEG_INFINITY as f32, f32::NEG_INFINITY);
+}
+
+fn ops() {
+    // f32 min/max
+    assert_eq((1.0 as f32).max(-1.0), 1.0);
+    assert_eq((1.0 as f32).min(-1.0), -1.0);
+    assert_eq(f32::NAN.min(9.0), 9.0);
+    assert_eq(f32::NAN.max(-9.0), -9.0);
+    assert_eq((9.0 as f32).min(f32::NAN), 9.0);
+    assert_eq((-9.0 as f32).max(f32::NAN), -9.0);
+
+    // f64 min/max
+    assert_eq((1.0 as f64).max(-1.0), 1.0);
+    assert_eq((1.0 as f64).min(-1.0), -1.0);
+    assert_eq(f64::NAN.min(9.0), 9.0);
+    assert_eq(f64::NAN.max(-9.0), -9.0);
+    assert_eq((9.0 as f64).min(f64::NAN), 9.0);
+    assert_eq((-9.0 as f64).max(f64::NAN), -9.0);
+
+    // f32 copysign
+    assert_eq(3.5_f32.copysign(0.42), 3.5_f32);
+    assert_eq(3.5_f32.copysign(-0.42), -3.5_f32);
+    assert_eq((-3.5_f32).copysign(0.42), 3.5_f32);
+    assert_eq((-3.5_f32).copysign(-0.42), -3.5_f32);
+    assert!(f32::NAN.copysign(1.0).is_nan());
+
+    // f64 copysign
+    assert_eq(3.5_f64.copysign(0.42), 3.5_f64);
+    assert_eq(3.5_f64.copysign(-0.42), -3.5_f64);
+    assert_eq((-3.5_f64).copysign(0.42), 3.5_f64);
+    assert_eq((-3.5_f64).copysign(-0.42), -3.5_f64);
+    assert!(f64::NAN.copysign(1.0).is_nan());
+}
+
 pub fn main() {
+    basic();
+    casts();
+    ops();
```

Comment:
Hm, `basic` and `ops` have nothing to do with casts, might be better to remove them here?